File: questao1/c/main.c
#include <stdio.h>

int main(void)
{
    char *s = "abcdabcbb";
    int max = 0;

    for (char *start = s; *start != '\0'; start++)
    {
        int count = 0;
        char seen[256] = {0}; // tabela ASCII

        printf("start: %c\n", *start);
        for (char *end = start; *end != '\0'; end++)
        {
            printf("end: %c\n", *end);
            printf("seen[%u] = %d\n", (unsigned char)*end,
                   seen[(unsigned char)*end]);

            if (seen[(unsigned char)*end])
                break; // caractere repetido encontrado
            seen[(unsigned char)*end] = 1;
            count++;
        }

        if (count > max)
            max = count;
    }

    printf("Maior substring sem repetição: %d caracteres\n", max);
    return 0;
}

File: questao1/lua/main.lua
-- main.lua
local s = "abcdabcbb"
local max = 0

for start = 1, #s do
    local count = 0
    local seen = {} -- tabela de caracteres vistos

    print("start: " .. s:sub(start, start))

    for i = start, #s do
        local ch = s:sub(i, i)
        print("end: " .. ch)
        print("seen[" .. string.byte(ch) .. "] = " .. tostring(seen[ch] and 1 or 0))

        if seen[ch] then
            break
        end

        seen[ch] = true
        count = count + 1
    end

    if count > max then
        max = count
    end
end

print("Maior substring sem repetição: " .. max .. " caracteres")
